api/v1alpha1/oadp_types.go

```diff
@@ -146,10 +146,29 @@ type ResticConfig struct {
 	PodConfig *PodConfig `json:"podConfig,omitempty"`
 }
 
+type NodeAgentConfig struct {
+	// Embedding ResticConfig
+	// +kubebuilder:validation:Optional
+	ResticConfig `json:",inline"`
+
+	// The type of uploader to transfer the data of pod volumes, the supported values are 'restic' or 'kopia'
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=restic;kopia
+	UploaderType string `json:"uploaderType,omitempty"`
+}
+
 // ApplicationConfig defines the configuration for the Data Protection Application
 type ApplicationConfig struct {
 	Velero *VeleroConfig `json:"velero,omitempty"`
+	// (deprecation warning) ResticConfig is the configuration for restic server.
```

Comment:
OpenAPI has a deprecated field, do not know if this works for this case https://github.com/kubernetes-sigs/kubebuilder/issues/2116#issuecomment-1050509977

Reply:
No it does not. The deprecation woks on the entire CRD and not one field.